### PR TITLE
[corlib] Import System.IO.TextReader/StreamReader from CoreFX

### PR DESCRIPTION
**Pull request by alexischr**

### CODE REVIEW

File: src/Common/src/CoreLib/System/IO/StreamReader.cs [hunk 1]
```diff
@@ -1018,7 +1018,11 @@ public override Task<int> ReadAsync(char[] buffer, int index, int count)
 
             CheckAsyncTaskInProgress();
 
+#if !MONO
```
Comment (marek-safar) [inline src/Common/src/CoreLib/System/IO/StreamReader.cs]: Why is this necessary?

File: src/Common/src/CoreLib/System/IO/TextReader.cs [hunk 2]
```diff
@@ -261,15 +265,29 @@ public virtual Task<int> ReadAsync(char[] buffer, int index, int count)
                     return t.Item1.Read(t.Item2.Span);
                 }, Tuple.Create(this, buffer), cancellationToken, TaskCreationOptions.DenyChildAttach, TaskScheduler.Default));
 
+#if !MONO
```
Comment (marek-safar) [inline src/Common/src/CoreLib/System/IO/TextReader.cs]: We want to use ValueTask, please undo this change

Comment (alexischr) [reply src/Common/src/CoreLib/System/IO/TextReader.cs]: The signature change is because StreamReader overrides this with internal `override async ValueTask<int> ReadAsyncInternal` which is not supported on MCS. Maybe we can make a MCS compilation define?

Comment (marek-safar) [reply src/Common/src/CoreLib/System/IO/TextReader.cs]: We have it (for very long time), it's called `__MonoCS__`
